### DISCUSSION TITLE
Failed to install the following SDK components

### DISCUSSION THREAD

**Discussion by Dario-Maselli** [codemagic-ci-cd/codemagic-docs#2410] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Run a build
2. Build Fails with:
```
Could not determine the dependencies of task ':app:packageStagingReleaseBundle'.
> Could not create task ':app:minifyStagingReleaseWithR8'.
   > Failed to install the following SDK components:
         platforms;android-34 Android SDK Platform 34
     Install the missing components using the SDK manager in Android Studio.```

### Expected results

I expect the app to build... There was no change from previous build.

### Actual results

 Build Fails with:
```
Could not determine the dependencies of task ':app:packageStagingReleaseBundle'.
> Could not create task ':app:minifyStagingReleaseWithR8'.
   > Failed to install the following SDK components:
         platforms;android-34 Android SDK Platform 34
     Install the missing components using the SDK manager in Android Studio.```

### Build id (optional)

65ef1432a489f1589fbc773c

**Comment by Dario-Maselli** [comment]
Sidenote: Why is the support chat no longer visible within CodeMagic?

**Reply by dtrdic** [reply]
Hello @Dario-Maselli based on the error message, it does look like the build failed to download the required assets from Shorebird. However, this could be a temporary issue. Can you please try re-running the build to see if it still results in the same error?
Regarding the support chat widget, it seems there are some issues at the moment and the team is already looking into it. 
I will let you know once the issue is resolved.

**Reply by dtrdic** [reply]
The issue with the chat widget not showing has been resolved as well. Please let us know in case you are still experiencing the same issue

**Comment by Reedyuk** [comment]
@dtrdic - hi, we are also getting this issue now, feels like a potential issue on codemagic's side?

**Reply by Reedyuk** [reply]
```

Caused by: java.util.zip.ZipException: Archive is not a ZIP archive
	at org.apache.commons.compress.archivers.zip.ZipFile.positionAtEndOfCentralDirectoryRecord(ZipFile.java:1221)
	at org.apache.commons.compress.archivers.zip.ZipFile.positionAtCentralDirectory(ZipFile.java:1097)
	at org.apache.commons.compress.archivers.zip.ZipFile.populateFromCentralDirectory(ZipFile.java:713)
	at org.apache.commons.compress.archivers.zip.ZipFile.<init>(ZipFile.java:376)
	... 197 more
"Install Android SDK Platform 34 (revision 3)" failed.

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':sample:compileDebugJavaWithJavac'.
> Could not determine the dependencies of null.
   > Failed to install the following SDK components:
         platforms;android-34 Android SDK Platform 34
     Install the missing components using the SDK manager in Android Studio.
```

**Reply by blasko** [reply]
We're experiencing this as well at the moment.

**Reply by dtrdic** [reply]
Seems there were connectivity issues or a temporary network glitch when fetching the Android SDK, but everything seems to be back to normal. @blasko can you also confirm if that’s the case? I already checked @Reedyuk's case over the Intercom chat

**Reply by blasko** [reply]
@dtrdic Everything is good on our side now, seems like a temp glitch to me as well.

**Comment by emmett-deen** [comment]
Received this issue 3 builds in a row so it doesn't seem to be temporary. Here is our latest build id: 65f359f345d5fefa6014e7ef

**Reply by dtrdic** [reply]
Hey @emmett-deen, I noticed that the build ID you shared is from yesterday. In your most recent build, the error message appears to be different. Nevertheless, we are actively monitoring this issue and implementing some improvements to prevent it from occurring again. But please let us know if you encounter this issue again in your newer builds

**Comment by Bonsaikitt3n** [comment]
This is happening to us starting this morning.

https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/8.1.0/gradle-8.1.0.jar'.
         > Premature end of Content-Length delimited message body (expected: 10,817,071; received: 2,621,424)

Packages are not being completed and Android 34 can't be loaded. We cleard deps caches. Builds fine locally, all packages are pulled down just fine.

**Comment by jeroenbloom** [comment]
We're also seeing this happening when building on the latest Flutter beta (`3.21.0-0.0.pre`). I created [an issue ticket on Discord](https://discord.com/channels/1131597315707261018/1219256779901239336) and back-linked this thread.

Build ID: `65f4753beee66e4973f4b8a5`

**Reply by dtrdic** [reply]
Hi @jeroenbloom, I can see that your latest builds were successful, and the buildID you sent was from 15.03., which aligns with errors reported by other users on that date. Since then, we have implemented some updates to prevent this issue from occurring again. But please let us know if the issue resurfaces

**Reply by jeroenbloom** [reply]
Hey there, thanks! I forgot to update this thread, sorry about that.

I temporary fixed it by installing the dependencies myself, though this may have coincided with the fixes on your end. I'll remove my extra build script and we'll see from there. Thanks for the support, I'll close my Discord ticket!

**Comment by Bonsaikitt3n** [comment]
This is happening again today. It fixed itself last week but our builds are failing again:


> Could not download gradle-7.3.1.jar (com.android.tools.build:gradle:7.3.1)
   > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/7.3.1/gradle-7.3.1.jar'.
      > Premature end of Content-Length delimited message body (expected: 9,320,290; received: 8,912,832)

**Reply by dtrdic** [reply]
@Bonsaikitt3n can you please share a build id as well so i can check it further? You can find it on the build overview page

**Reply by Bonsaikitt3n** [reply]
Happening today off and on: Build : 65fb23f2c0c9136f66e6306b , 65fb241813c57fa34c55e251

**Comment by Bonsaikitt3n** [comment]
It resolved around 12PM Pst yesterday, today every build failing. Here is one of the build ids 65fc5980c6a6a27511f32c7d

**Reply by Dario-Maselli** [reply]
@Bonsaikitt3n - Have a look here: https://status.codemagic.io/incidents

**Comment by softlion** [comment]
I fixed it by adding that environment section, and the below script:

```yaml
    environment:
      java: 21         # Define default, or platform version (e.g. 11)
#      ndk: r26c       # Does not work, the wrong tool is invoked
```


```shell
$ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager --update
$ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager --install "platforms;android-34"

ls           ls -alF $ANDROID_HOME/platforms
```

Btw impossible to install an ndk with environment. An obsolete version of sdkmanager is invoked (in another android sdk folder).

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion due to a lack of activity. You can open the discussion again by replying to this message or opening a new discussion.

**Reply by softlion** [reply]
Did you fixed it on your side ? Because that's what this discussion is about.

**Reply by dvmatyun** [reply]
Got the issue in build id 6667427c077ad83ab6b06610

**Comment by dvmatyun** [comment]
Got the issue in build id 6667427c077ad83ab6b06610
